src/components/general/Collapser.js

```diff
@@ -1,55 +1,64 @@
-/* @flow */
-import type {Element as ReactElement} from "React"
-import React, { useState } from "react"
-import type {Node as ReactNode} from "react"
-
-import classnames from "classnames"
-import Chevron from "../../icons/Chevron"
-import Button from "../base/Button"
-import type {AnalyticsEvent} from "../../google-tag-manager"
+import React, { useState, useEffect } from "react";
+import type { Element as ReactElement, Node as ReactNode } from "react";
+import classnames from "classnames";
+import Chevron from "../../icons/Chevron";
+import Button from "../base/Button";
+import type { AnalyticsEvent } from "../../google-tag-manager";
 
 type Props = {
     className?: string,
+    header?: string,
     expandMessage: string,
     collapseMessage?: string,
     initiallyExpanded?: boolean,
     contentPreview?: ReactNode,
     children?: ReactNode,
     onClick?: Function,
-    analyticsEvent?: AnalyticsEvent
-}
+    analyticsEvent?: AnalyticsEvent,
+    onToggle?: (isCollapsed: boolean) => void,
+    externalCollapsed?: boolean,
+    icon?: ReactNode,
+};
 
-// By default, if 'collapseMessage' is not defined, then the component will not
-// display a button to collapse once it has been expanded.
 function Collapser({
     className,
+    header,
     expandMessage,
     collapseMessage,
     initiallyExpanded = false,
```

Comment:
I know this wasn't added in this PR, but `initiallyExpanded` could be removed. I see it's only used here 
https://github.com/ask-izzy/ask-izzy/blob/f3c60c3f39b16ec931607db540bb6096afff1763/src/components/DebugPersonalisation.js#L64

and here https://github.com/ask-izzy/ask-izzy/blob/f3c60c3f39b16ec931607db540bb6096afff1763/src/components/DebugContainer.js#L12

Because you have the `externalCollapsed` prop the `initiallyExpanded` becomes redundant because `externalCollapsed` will still trigger the useEffect on-load.

``` js
useEffect(() => {
        // Runs this
        if (typeof externalCollapsed === "boolean" && externalCollapsed !== isCollapsed) {
            setIsCollapsed(externalCollapsed);
        }
    }, [externalCollapsed]); // Triggered on load
```

This means the initial value will be set like this and not defined by the prop
`const [isCollapsed, setIsCollapsed] = useState(true);` 

And this part of the debug container will look like this

``` jsx
<DebugContainer
            message="Debug ISS Query"
            externalCollapsed={
                !!Storage.getJSON("issParamsOverride")
            }
        >
```

Here are all the places I've found that use this prop
https://github.com/search?q=repo%3Aask-izzy%2Fask-izzy%20initiallyExpanded&type=code

Reply:
Cheers Trent! this completely slipped my mind as I was trying out different things to get this collapser working, removed it now from the code so waiting for that to build